### DISCUSSION TITLE
How to change/edit a static field value

### DISCUSSION THREAD

**Discussion by KWINEKO** [vfsfitvnm/frida-il2cpp-bridge#313] [Q&A] [answered]
```
// API
class WebAPI.Protocol : System.Object
{
    static System.String address= "http://localhost:8002";
    ......
    static WebAPI.IContext DevelopmentCatalog();
    static System.Void OnDevelopmentCatalogCallback(WebAPI.IContext context);
    static WebAPI.IContext DevelopmentGet(System.Int64 recordKey);
    static System.Void OnDevelopmentGetCallback(WebAPI.IContext context);
    ......
}
```
Hello, I'm trying to change/edit fields in a class.
I read the discussion in #254, but is not for the static field.
If I edit the value directly, then will cause an error.
```
Protocol.field<Il2Cpp.String>("address").value = Il2Cpp.string("http://localhost:8024")
```
The error:
```
il2cpp: cannot write the value of field address as it's thread static or literal
    at set value (il2cpp/structs/field.ts:96)
    at <anonymous> (agent/index.ts:34)
    at <anonymous> (il2cpp/structs/thread.ts:92)
    at call (native)
    at <anonymous> (il2cpp/structs/method.ts:328)
```

Is there any way to modify thread static fields?

**Comment by vfsfitvnm** [comment, marked as answer]
In theory, *yes*, there is.
Unfortunately I couldn't find a decent/working way to do that, yet.

However, you could do some hardcoding and you'd be fine. Take a look here https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/ae7ef9ea2ca0e33fe50994179b2a4751541549e4/src/il2cpp/structs/thread.ts#L119 to see how I look for thread local values

**Reply by KWINEKO** [reply]
Thank you, I roughly understand how you get the local value by Frida.

I have temporarily chosen to modify "global metadata.dat" to implement it. 
Because I don't know how to modify the value while the program is running.
 (I probably need to learn more about Frida hook)
